src/main/java/objects/battle/enemy/factory/EnemyFactory.java

```diff
@@ -0,0 +1,72 @@
+package objects.battle.enemy.factory;
+
+import database.managers.EnemyDataManager;
+import database.managers.GimmickDataManager;
+import database.objects.EnemyData;
+import database.objects.GimmickData;
+import objects.battle.enemy.EnemyInfo;
+import objects.battle.enemy.ai.EnemyAI;
+import objects.battle.enemy.gimmick.*;
+import objects.character.Boss;
+import objects.character.Enemy;
+
+
+public class EnemyFactory {
+
+    /**
+     * This method returns the enemy or boss instance using the information given by database
+     * This method returns Enemy so if the boss is returned, make sure you cast, so that the object
+     * is Boss.
+     * @param name of the enemy to create
+     * @return instance of enemy
+     */
+    public static Enemy createEnemy(String name) throws Exception {
+        EnemyInfo enemyInfo = EnemyInfoFactory.createEnemyInfo(name);
+        EnemyAI enemyAI = EnemyAIFactory.createEnemyAI(name);
+        EnemyDataManager data = new EnemyDataManager();
```

Comment:
You should not be creating database managers everytime you call a method. Instead, create a private instance of a database at the highest level of the class hierarchy and pass the database through as a parameter when calling a method.